java/memory/src/main/java/org/apache/arrow/memory/ReferenceManager.java

```diff
@@ -0,0 +1,101 @@
+package org.apache.arrow.memory;
+
+import io.netty.buffer.ArrowBuf;
+
+/**
+ * Reference Manager manages one or more ArrowBufs that share the
+ * reference count for the underlying memory chunk.
+ */
+public interface ReferenceManager {
+
+  /**
+   * Return the reference count
+   * @return reference count
+   */
+  int getRefCount();
+
+  /**
+   * Decrement this reference manager's reference count for the associated underlying
+   * memory. If the reference count drops to 0, it implies that ArrowBufs managed by this
+   * reference manager no longer need access to the underlying memory
+   * @return the new reference count
```

Comment:
it returns a boolean though?

Reply:
fixed.

Reply:
yes the interface in old ArrowBuf had the semantics that it will return true if ref count has dropped to 0, false otherwise.